lib/AST/NameLookup.cpp

```diff
@@ -1448,15 +1450,38 @@ DirectLookupRequest::evaluate(Evaluator &evaluator,
 
     Table.updateLookupTable(decl);
   } else if (!Table.isLazilyComplete(name.getBaseName())) {
-    // The lookup table believes it doesn't have a complete accounting of this
-    // name - either because we're never seen it before, or another extension
-    // was registered since the last time we searched. Ask the loaders to give
-    // us a hand.
     DeclBaseName baseName(name.getBaseName());
-    populateLookupTableEntryFromLazyIDCLoader(ctx, Table, baseName, decl);
-    populateLookupTableEntryFromExtensions(ctx, Table, baseName, decl);
 
-    Table.markLazilyComplete(baseName);
+    // If we're building libSyntaxParser, #if out the two clang importer
+    // requests because libSyntaxParser doesn't know about the clang importer.
+    if (decl->getClangDecl() &&
+        isa<clang::NamespaceDecl>(decl->getClangDecl())) {
```

Comment:
Nit: You can use `isa_and_nonnull` for these tests.

Reply:
:) thanks, I've needed that several times.